views/ViewChart.react.js

```diff
@@ -1,26 +1,54 @@
-import React from 'react'
+import React, { useState } from 'react'
 import { connect } from 'react-redux'
 import { compose } from 'redux'
 import { withStyles } from '@material-ui/core/styles'
-
-import { Typography } from '@material-ui/core'
+import { Button, Drawer, Typography } from '@material-ui/core'
+import FilterListIcon from '@material-ui/icons/FilterList'
 import { chartStyles } from './styles/chart_styles'
 import AppLayout from './layouts/AppLayout'
 import BarGraph from './components/BarGraph'
 import GraphTable from './components/GraphTable'
+import ChartFilterForm from './forms/CharFilterForm'
+import { routes } from './routes/routes'
 
 const ViewChart = ({ graph, classes }) => {
-  const { title, description } = graph
+  const { title, description, filters } = graph
+  const [showFilterMenu, setFilterMenuToggle] = useState(false)
+  const openFilterMenu = () => setFilterMenuToggle(true)
+  const onCloseFilterMenu = () => setFilterMenuToggle(false)
+  const handleSubmit = (e, updatedFilters) => {
+    e.preventDefault()
+    window.location.assign(
+      routes.chart(graph.chart_id, { filters: updatedFilters })
+    )
+  }
   return (
     <AppLayout title={title}>
       {description && (
-        <Typography variant="body1" gutterBottom>
-          <pre style={{ fontFamily: 'inherit' }}>{description}</pre>
-        </Typography>
+        <div className={classes.viewChartRow}>
+          <Typography variant="subtitle1">
+            <pre style={{ fontFamily: 'inherit' }}>{description}</pre>
+          </Typography>
+          <Button
+            variant="outline"
+            color="default"
+            className={classes.showFilterButton}
+            onClick={() => openFilterMenu()}
+          >
+            <FilterListIcon className={classes.leftIcon} />
+            Filters
+          </Button>
+        </div>
       )}
-
+      <Drawer anchor="left" open={showFilterMenu} onClose={onCloseFilterMenu}>
+        <ChartFilterForm
+          initialValues={filters}
+          onSubmit={handleSubmit}
+          classes={classes}
+        />
+      </Drawer>
       <BarGraph graph={graph} classes={classes} />
-      <GraphTable graph={graph} />
+      {!!graph.dataBySite.length ?? <GraphTable graph={graph} />}
```

Comment:
why is this needed? I don't think ?? is what we want here since the left side will never be null or undefined (it will be a boolean). So in this case I believe the graph table is never rendered?

Reply:
Thank you. This is what it's supposed to be. https://github.com/AMP-SCZ/dpdash/pull/248/commits/1a688fa3ed46d74c7d2545e89cc2c27c0130f3fe

If we don't use this then the page looks like this 
<img width="1178" alt="Screen Shot 2023-01-11 at 4 05 40 PM" src="https://user-images.githubusercontent.com/19805355/211929020-4b17f5ea-d2a2-4e65-8dff-8720326dd737.png">

Reply:
Now it looks like this.
<img width="1184" alt="Screen Shot 2023-01-11 at 5 24 09 PM" src="https://user-images.githubusercontent.com/19805355/211939520-0d60f178-ef9c-498a-ad16-4a7da765accb.png">